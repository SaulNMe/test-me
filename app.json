{
  "expo": {
    "name": "Qualificame",
    "slug": "qualificame-native",
    "privacy": "public",
    "sdkVersion": "33.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splashCustom.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff",

    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    }
  }
}
