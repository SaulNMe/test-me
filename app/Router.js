import React, { Component } from 'react';
import { Platform, Animated, Easing } from 'react-native';
import { createAppContainer, createStackNavigator, createBottomTabNavigator, createSwitchNavigator } from 'react-navigation';
import WelcomeScreen from 'qualificame-native/app/screens/WelcomeScreen';
import LoginScreen from 'qualificame-native/app/screens/LoginScreen';
import SignUpScreen from 'qualificame-native/app/screens/SignUpScreen';
import CreateAccountScreen from 'qualificame-native/app/screens/CreateAccountScreen';
import SelectPlanScreen from 'qualificame-native/app/screens/SelectPlanScreen';
import PaymentScreen from 'qualificame-native/app/screens/PaymentScreen';
import KiosksScreen from 'qualificame-native/app/screens/KiosksScreen';
import CreateKioskScreen from 'qualificame-native/app/screens/CreateKioskScreen';
import KioskDetailScreen from 'qualificame-native/app/screens/KioskDetailScreen';
import KioskConfigurationScreen from 'qualificame-native/app/screens/KioskConfigurationScreen';
import KioskContactDataScreen from 'qualificame-native/app/screens/KioskContactDataScreen';
import KioskOptionsScreen from 'qualificame-native/app/screens/KioskOptionsScreen';
import ReportsScreen from 'qualificame-native/app/screens/ReportsScreen';
import CreateReportScreen from 'qualificame-native/app/screens/CreateReportScreen';
import SummaryReportScreen from 'qualificame-native/app/screens/SummaryReportScreen';
import ReportDetailScreen from 'qualificame-native/app/screens/ReportDetailScreen';
import MailBoxScreen from 'qualificame-native/app/screens/MailBoxScreen';
import MyAccountScreen from 'qualificame-native/app/screens/MyAccountScreen';
import ProPlanScreen from 'qualificame-native/app/screens/ProPlanScreen';
import EnterprisePlanScreen from 'qualificame-native/app/screens/EnterprisePlanScreen';
import ChangePasswordScreen from 'qualificame-native/app/screens/ChangePasswordScreen';
import FaqScreen from 'qualificame-native/app/screens/FaqScreen';
import AlertsScreen from 'qualificame-native/app/screens/AlertsScreen';
import BottomBar from 'qualificame-native/app/components/BottomBar';
import LoadingScreen from 'qualificame-native/app/screens/LoadingScreen';
import SendEmailScreen from 'qualificame-native/app/screens/SendEmailScreen';
import VerificationCodeScreen from 'qualificame-native/app/screens/VerificationCodeScreen';
import ConfirmPasswordScreen from 'qualificame-native/app/screens/ConfirmPasswordScreen';
import AuthLoadingScreen from 'qualificame-native/app/screens/AuthLoadingScreen';
import ScannerScreen from 'qualificame-native/app/screens/ScannerScreen';

import GetStateForAction from 'qualificame-native/app/utils/GetStateForAction';
import TransitionConfiguration from 'qualificame-native/app/utils/TransitionConfiguration';

const RecoverPasswordStack = createStackNavigator({
	SendEmailScreen,
	VerificationCodeScreen,
	ConfirmPasswordScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
})

const AuthModalStackNavigator = createStackNavigator({
	WelcomeScreen,
	LoginScreen,
	RecoverPasswordStack,
	SignUpScreen,
	CreateAccountScreen,
	SelectPlanScreen,
	PaymentScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration
});

const KioskConfigurationStack = createStackNavigator({
	KioskConfigurationScreen,
	KioskOptionsScreen,
	KioskContactDataScreen
});

const CreateReportStack = createStackNavigator({
	CreateReportScreen,
	SummaryReportScreen,
	LoadingScreen,
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
});

GetStateForAction(CreateReportStack);

const BottomBarNavigator = createBottomTabNavigator({
	KiosksScreen,
	ReportsScreen,
	MailBoxScreen,
	MyAccountScreen
}, {
	initialRouteName: 'KiosksScreen',
	tabBarComponent: BottomBar
});

const CameraModalNavigator = createStackNavigator({
	ScannerScreen
}, {
	mode: 'modal',
	navigationOptions: {
		header: null,
	}
})

const MainStackNavigator = createStackNavigator({
	BottomBarNavigator: {
		screen: BottomBarNavigator,
		navigationOptions: {
			header: null,
		}
	},
	KioskDetailScreen,
	CameraModalNavigator,
	ReportDetailScreen,
	ProPlanScreen,
	EnterprisePlanScreen,
	ChangePasswordScreen,
	FaqScreen
}, {
	initialRouteName: 'BottomBarNavigator',
});

const ModalStack = createStackNavigator({
	MainStackNavigator,
	AlertsScreen,
	CreateKioskScreen,
	KioskConfigurationStack,
	CreateReportStack
}, {
	mode:'modal',
	headerMode: 'none'
});

const MainNavigation = createSwitchNavigator({
	AuthLoading: AuthLoadingScreen,
	Auth: AuthModalStackNavigator,
	App: ModalStack,
}, {
	initialRouteName: 'AuthLoading'
});

export default createAppContainer(MainNavigation);




 




