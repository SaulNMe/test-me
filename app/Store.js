import { createStore, applyMiddleware, compose } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel1 from 'redux-persist/lib/stateReconciler/autoMergeLevel1';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk'
import rootReducer from 'qualificame-native/app/reducers'
import Reactotron from 'qualificame-native/config/ReactotronConfig'

const persistConfig = {
	key: 'root',
	storage: storage,
	stateReconciler: autoMergeLevel1,
	whitelist: ['Auth']
};

const pReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
	pReducer,
	compose(
		applyMiddleware(thunk),
		Reactotron.createEnhancer()
	)
);

export default store;
export const persistor = persistStore(store);