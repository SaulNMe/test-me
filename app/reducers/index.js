import Events, * as fromEvents from 'qualificame-native/app/reducers/EventsReducer.js';
import Auth, * as fromAuth from 'qualificame-native/app/reducers/AuthReducer.js';
import TokenRefresh, * as fromTokenRefresh from 'qualificame-native/app/reducers/TokenRefreshReducer.js'
import Kiosks, * as fromKiosks from 'qualificame-native/app/reducers/KiosksReducer.js';
import Reports, * as fromReports from 'qualificame-native/app/reducers/ReportsReducer.js';
import Choices, * as fromChoices from 'qualificame-native/app/reducers/ChoicesReducer.js';
import Alerts, * as fromAlerts from 'qualificame-native/app/reducers/AlertsReducer.js';

import { combineReducers } from 'redux'

const rootReducer = combineReducers({
    Auth,
    TokenRefresh,
    Kiosks,
    Reports,
    Events,
    Choices,
    Alerts,
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Token Refresh
export const getIsRefreshingToken = state =>
	fromTokenRefresh.getIsRefreshingToken(state.TokenRefresh);

export const getRefreshCall = state =>
	fromTokenRefresh.getRefreshCall(state.TokenRefresh);

//Kiosks
export const getKiosks = state =>
	fromKiosks.getKiosks(state.Kiosks);

export const getKioskById = (state, kioskId) =>
	fromKiosks.getKioskById(state.Kiosks, kioskId)

export const getIsLoadingKiosks = state =>
	fromKiosks.getIsLoadingKiosks(state.Kiosks);

export const getErrorKiosks = state =>
	fromKiosks.getErrorKiosks(state.Kiosks);

//Reports

export const getReports = state =>
	fromReports.getReports(state.Reports);

export const getIsLoadingReports = state =>
	fromReports.getIsLoadingReports(state.Reports);
export const getErrorReports = state =>
	fromReports.getErrorReports(state.Reports);

// Events
export const getEvents = state =>
	fromEvents.getEvents(state.Events);

export const getEventsGroupedByDay = (state, ids) =>
	fromEvents.getEventsGroupedByDay(state.Events, ids);

export const getSelectedMonth = state => 
	fromEvents.getSelectedMonth(state.Events);

export const getEventsSelectedDateRange = state => 
	fromEvents.getEventsSelectedDateRange(state.Events);

// Choices
export const getChoices = state =>
	fromChoices.getChoiceById(state.Choices);

// Alerts 
export const getAlerts = state =>
	fromAlerts.getAlerts(state.Alerts);

export const unReadAlerts = state =>
	fromAlerts.unReadAlerts(state.Alerts);