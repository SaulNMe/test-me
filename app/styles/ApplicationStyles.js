import Fonts from './Fonts'
import Colors from './Colors'
import Metrics from './Metrics'
import { StatusBar, Platform } from 'react-native';

// This file is for a reusable grouping of Theme items.
// Similar to an XML fragment layout in Android

const ApplicationStyles = {
	screen: {
		imageFill: {
			position:'absolute', 
			top: 0, 
			bottom: 0, 
			right: 0, 
			left: 0, 
			width:null, 
			height:null		
		},
		fill: {
			width: '100%',
			height: '100%'
		},
		container: {
			flex: 1
		},
		card: {
			marginVertical: Metrics.smallMargin,
			padding: Metrics.smallMargin,
			backgroundColor: Colors.white,
			shadowColor: '#000',
			shadowRadius: 4,
			shadowOpacity: 0.1,
			shadowOffset: { width: 0, height: 3 },
			elevation: 3,
			borderRadius: 5,
		},
		absolute: {
			position: 'absolute'
		},
		setToBot: {
			bottom: 0,
			right: 0,
			left: 0
		},
		behind: {
			zIndex: -1
		},
		headerImage: {
			width: Metrics.screenWidth,
			height: (Metrics.screenWidth * .46 ) 
		},
		footerImage: {
			width: Metrics.screenWidth,
			height: (Metrics.screenWidth * .52 ) 
		},
		customSpacer: {
			marginTop: (Metrics.screenWidth * .46 ) - (Metrics.navBarHeight + 27),
		},
		bottomView: {
			position:'absolute',
			bottom: 0,
			width: '100%',
			zIndex: 9999
		},
		bottomViewHeight: {
			height: 64
			//height 64 + 16margin
		},
		alert:{
			zIndex:9999, 
		},
	},
	margins: {
		doubleBaseMargin: {
			marginTop: Metrics.doubleBaseMargin
		},
		doubleBaseVerticalMargin: {
			marginVertical: Metrics.doubleBaseMargin
		},
		tinyVerticalMargin: {
			marginVertical: Metrics.tinyMargin
		},
		smallVerticalMargin: {
			marginVertical: Metrics.smallMargin
		},
		baseVerticalMargin: {
			marginVertical: Metrics.baseMargin
		},
		smallHorizontalMargin: {
			marginHorizontal: Metrics.smallMargin
		},
		baseHorizontalMargin: {
			marginHorizontal: Metrics.baseMargin
		},
		smallTopMargin: {
			marginTop: Metrics.smallMargin
		},	
		baseTopMargin: {
			marginTop: Metrics.baseMargin
		},
		tinyRightMargin: {
			marginRight: Metrics.tinyMargin
		},
		smallRightMargin: {
			marginRight: Metrics.smallMargin
		},
		smallLeftMargin: {
			marginLeft: Metrics.smallMargin
		},		
		baseLeftMargin: {
			marginLeft: Metrics.baseMargin
		},
		smallBottomMargin: {
			marginBottom: Metrics.smallMargin
		},
		baseBottomMargin: {
			marginBottom: Metrics.baseMargin
		},
		baseBottomPadding: {
			paddingBottom: Metrics.baseMargin
		},
		doubleBaseBottomPadding: {
			paddingBottom: Metrics.doubleBaseMargin,
		},
		statusBarSpace: {
			paddingTop: (Platform.OS != 'ios') ? StatusBar.currentHeight : 0
		},
		navBarHeight: {
			height: Metrics.navBarHeight,
		},
		baseTopPadding: {
			paddingTop: Metrics.baseMargin
		},
		tinyVerticalPadding: {
			paddingVertical: Metrics.tinyMargin
		},		
		smallVerticalPadding: {
			paddingVertical: Metrics.smallMargin
		},
		baseVerticalPadding: {
			paddingVertical: Metrics.baseMargin
		},
		smallHorizontalPadding: {
			paddingHorizontal: Metrics.smallMargin
		},
		baseHorizontalPadding: {
			paddingHorizontal: Metrics.baseMargin
		}

	}, 
	flexBox: {
		column: {
			flexDirection: 'column'
		},
		row: {
			flexDirection: 'row'
		},
		antirow: {
			flexDirection: 'row-reverse'
		},
		flex1: {
			flex: 1
		},
		flex2: {
			flex:2
		},
		flex: {
			flex:3
		},
		justifyContentFlexStart: {
			justifyContent: 'flex-start'
		},
		justifyContentFlexEnd: {
			justifyContent: 'flex-end'
		},
		justifyContentCenter: {
			justifyContent: 'center'
		},
		justifyContentSpaceBetween: {
			justifyContent: 'space-between'
		},
		justifyContentSpaceAround:{
			justifyContent: 'space-around'
		},
		justifyContentSpaceEvenly:{
			justifyContent: 'space-evenly'
		},
		alignItemFlexStart: {
			alignItems: 'flex-start'
		},
		alignItemsFlexEnd: {
			alignItems: 'flex-end'
		},
		alignItemsCenter: {
			alignItems: 'center'
		},
		alignItemsStretch: {
			alignItems: 'stretch'
		},
		centerObjects: {
			justifyContent: 'center',
			alignItems: 'center'
		},
		alignSelfCenter: {
			alignSelf: 'center'
		},
		alignSelfFlexStart: {
			alignSelf: 'flex-start'
		},
		wrap: {
			flexWrap: 'wrap'
		},
		alignContentFlexStart: {
			alignContent: 'flex-start'
		},
		alignContentFlexEnd: {
			alignContent: 'flex-end'
		},
		alignContentCenter: {
			alignContent: 'center'
		},
		alignContentSpaceBetween: {
			alignContent: 'space-between'
		},
		alignContentSpaceAround: {
			alignContent: 'space-around'
		},
		fullWidth: {
			width: '100%'
		},
		fullHeigth: {
			height: '100%'
		},
		deviceHeight: {
			height: Metrics.screenHeight
		}
	},
	stackNavigatorOptions: {
		removeHeader: {
			header: null
		},
		transparentHeader: {
			headerTransparent: true,
			headerStyle: {
				backgroundColor: Colors.transparent,
				elevation: 0, // remove shadow on Android
				shadowOpacity: 0, // remove shadow on iOS
				borderBottomWidth: 0,
			},
			gesturesEnabled: false,
			headerTintColor: Colors.white
		},
		bottomTab: {
			activeTintColor: Colors.blue,
			inactiveTintColor: Colors.light,
			style: {
				backgroundColor: Colors.white
			}
		}
	},
	navigationOptions: {
		headerTransparent: true,
		headerStyle: {
			backgroundColor: Colors.transparent,
			elevation: 0, // remove shadow on Android
			shadowOpacity: 0, // remove shadow on iOS
			borderBottomWidth: 0,
		},
		gesturesEnabled: false,
	}
} 

export default ApplicationStyles
