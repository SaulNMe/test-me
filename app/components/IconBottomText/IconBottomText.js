import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './IconBottomTextStyle';

import { Feather } from '@expo/vector-icons';
import TitleText from 'qualificame-native/app/components/TitleText';
import { Colors } from 'qualificame-native/app/styles';

export default class IconBottomText extends PureComponent {
	render() {
		return (
			<View style={[styles.centerObjects, {width: 255}]}>
				<View style={[styles.bubbleForm, {backgroundColor: Colors[this.props.color], borderColor: Colors[this.props.color]}]}>
					<Feather name={this.props.iconName} size={70} color={this.props.iconColor}/>
				</View>
				<View> 
					<TitleText
						text= {this.props.title}
						color= {this.props.color}
						align="center"
					/>
				</View>
			</View>
		);
	}
}

	IconBottomText.propTypes = {
		iconColor: PropTypes.string,
		iconName: PropTypes.string,
		color: PropTypes.string,
		title: PropTypes.string,
	}

	IconBottomText.defaultProps = {
		iconColor: 'white',
		iconName: 'award',
		color: 'red',
		title: ''
	}
