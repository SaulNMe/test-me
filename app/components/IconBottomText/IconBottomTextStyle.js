import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	...ApplicationStyles.margins,
	...ApplicationStyles.flexBox,
	bubbleForm: {
		width: 100,
		height: 100,
		borderRadius: 90,
		justifyContent: 'center',
		alignItems: 'center',
		borderWidth: 1
	},
});
