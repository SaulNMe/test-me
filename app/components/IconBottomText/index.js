import IconBottomText from './IconBottomText';

export default IconBottomText;
