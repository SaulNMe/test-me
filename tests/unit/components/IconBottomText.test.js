import React from 'react';
import { shallow } from 'enzyme';

import IconBottomText from 'qualificame-native/app/components/IconBottomText';

import Feather from 'react-native-vector-icons/Feather';
import TitleText from 'qualificame-native/app/components/TitleText';
import { Colors } from 'qualificame-native/app/styles';
import { Text, View } from 'react-native';

function createTestProps (props) {
	return {
		// common props
		iconColor: 'blue',
		iconName: 'moon',
		color: 'red',
		title: 'test title',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<IconBottomText {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('displays test title as expected', () => {
		expect(wrapper.find(TitleText).prop('text')).toEqual('test title');
	});
	it('displays TitleText red color', () => {
		expect(wrapper.find(TitleText).prop('color')).toEqual('red');
	});
	it('displays the correct feather icon ', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
		expect(wrapper.find(Feather).props().name).toEqual('moon')
	});
	it('displays the correct feather icon color', () => {
		expect(wrapper.find(Feather).exists()).toBeTruthy();
		expect(wrapper.find(Feather).props().color).toEqual('blue')
	});	
});