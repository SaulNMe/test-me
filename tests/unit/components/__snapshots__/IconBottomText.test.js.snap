// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "width": 255,
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 90,
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "width": 100,
        },
        Object {
          "backgroundColor": "#EB3F3F",
          "borderColor": "#EB3F3F",
        },
      ]
    }
  >
    <Icon
      allowFontScaling={false}
      color="blue"
      name="moon"
      size={70}
    />
  </Component>
  <Component>
    <TitleText
      align="center"
      color="red"
      text="test title"
      weight=""
    />
  </Component>
</Component>
`;
